transmission/transmission_test.go

```diff
@@ -820,6 +823,100 @@ func TestFireBatchWithBrokenFirstEvent(t *testing.T) {
 	})
 }
 
+type fakeBatch struct {
+	block <-chan struct{}
+	send  chan<- []interface{}
+	items []interface{}
+}
+
+func (fb *fakeBatch) Add(item interface{}) {
+	fb.items = append(fb.items, item)
+}
+
+func (fb *fakeBatch) Fire(notifier muster.Notifier) {
+	defer notifier.Done()
+	<-fb.block
+	fb.send <- fb.items
+}
+
+func TestHoneycombTransmissionFlush(t *testing.T) {
+	ev := &Event{
+		Metadata: "adder",
+	}
+
+	t.Run("Flush should send all data that has been Added", func(t *testing.T) {
+		w := new(Honeycomb)
+		w.MaxBatchSize = 1000
+		w.PendingWorkCapacity = 1
+		block := make(chan struct{})
+		close(block) // don't block
+		sendChan := make(chan []interface{})
+		b := &fakeBatch{
+			send:  sendChan,
+			block: block,
+		}
+
+		batchCount := int32(0)
+		w.batchMaker = func() muster.Batch {
+			t.Logf("creating batch %d", atomic.LoadInt32(&batchCount))
+			if atomic.CompareAndSwapInt32(&batchCount, 0, 1) {
+				return b
+			}
+			return &fakeBatch{}
+			// We want to be sure that the data we enqueue is flushed in the batch we
+			// expect. By not having s send channel on subsequent batches, we'll
+			// panic if we flush to other batches.
+		}
+
+		if err := w.Start(); err != nil {
+			t.Error("unable to start", err)
+		}
+		defer w.Stop()
+		w.Add(ev)
+		go func() {
+			if err := w.Flush(); err != nil {
+				t.Error("unable to flush", err)
+			}
+		}()
```

Comment:
is there anything that is going to block and ensure that this goroutine finishes before the test run itself finishes?

Reply:
Nothing does this directly, but I don't think that's needed here: We just want to check that calling flush did indeed cause the batch to get flushed. The error check here is really just to help to debug a test failure.